crates/sui-framework/sources/coin.move

```diff
@@ -224,6 +224,17 @@ module sui::coin {
         )
     }
 
+    /// Split coin `self` into `n` coins with equal balances, or as close as
+    /// possible, if the balance is not evenly divisible by `n`.
+    public entry fun split_n<T>(self: &mut Coin<T>, n: u64, ctx: &mut TxContext) {
```

Comment:
For `n = 0`, I think we either want to fail with an appropriate error-code (preferred) or no-op. Otherwise, we'll fail with an underflow error at 232.